packages/validator-ajv8/src/validator.ts

```diff
@@ -137,12 +138,17 @@ export default class AJV8Validator<T = any, S extends StrictRJSFSchema = RJSFSch
         this.ajv.addSchema(rootSchema, rootSchemaId);
       }
       const schemaWithIdRefPrefix = withIdRefPrefix<S>(schema) as S;
+      const schemaHash = hashForSchema(schemaWithIdRefPrefix);
       let compiledValidator: ValidateFunction | undefined;
       if (schemaWithIdRefPrefix[ID_KEY]) {
         compiledValidator = this.ajv.getSchema(schemaWithIdRefPrefix[ID_KEY]);
+      } else {
+        compiledValidator = this.ajv.getSchema(schemaHash);
       }
       if (compiledValidator === undefined) {
-        compiledValidator = this.ajv.compile(schemaWithIdRefPrefix);
+        compiledValidator =
+          this.ajv.addSchema(schemaWithIdRefPrefix, schemaHash).getSchema(schemaHash) ||
+          this.ajv.compile(schemaWithIdRefPrefix);
```

Comment:
I'm curious about the `||` of the two statements. Why not do this instead?
```suggestion
        const addedSchema = this.ajv.addSchema(schemaWithIdRefPrefix, schemaHash).getSchema(schemaHash);
        compiledValidator = this.ajv.compile(addedSchema);
```

Reply:
This is recommended by the ajv docs https://ajv.js.org/guide/managing-schemas.html#pre-adding-all-schemas-vs-adding-on-demand.

`compile` will cache by the subschema `$id` property, or by the object hash. It doesn't allow specifically storing and fetching with a specific hash. So the `||` allows us to store and fetch by the hash, and then fallback to the default behavior if necessary.

Reply:
Can you add the link to the documentation above this line then?